stingray/modeling/parameterestimation.py

```diff
@@ -233,7 +251,11 @@ def fit(self, lpost, t0, neg=True, scipy_optimize_options=None):
             raise TypeError("lpost must be a subclass of "
                             "Posterior or LogLikelihoood.")
 
-        if not len(t0) == lpost.npar:
+        newmod = lpost.model.copy()
+        newmod.parameters = t0
+        p0, _ = _model_to_fit_params(newmod)
```

Comment:
I'm not sure I understand what these three lines do? Why are you copying the `lpost` object?